File: app/src/main/java/com/xingen/hookdemo/hook/service/ProxyService.java
package com.xingen.hookdemo.hook.service;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import androidx.annotation.Nullable;

/**
 * @author HeXinGen
 * date 2019/6/13.
 */
public class ProxyService extends Service {
    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        ServiceHookManager.startService( intent, flags,  startId);
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}

File: app/src/main/java/com/xingen/hookdemo/hook/service/ProxyService.kt
package com.xingen.hookdemo.hook.service

import android.app.Service
import android.content.Intent
import android.os.IBinder

/**
 * @author HeXinGen
 * date 2019/6/13.
 */
class ProxyService : Service() {

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        ServiceHookManager.startService(intent, flags, startId)
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }
}
